platform/darwin/src/MGLMapSnapshotter.mm

```diff
@@ -74,6 +75,7 @@ - (instancetype)initWithImage:(nullable MGLImage *)image scale:(CGFloat)scale po
         _latLngForFn = std::move(latLngForFn);
         _scale = scale;
         _image = image;
+        [MGLMapboxEvents pushTurnstileEvent];
```

Comment:
I think this should be in `-startWithQueue:completionHandler:` - an `MGLMapSnapshotter` may be allocated and not used, or used multiple times.

Perhaps sending the turnstile event somewhere around 

https://github.com/mapbox/mapbox-gl-native/blob/53351c2ae83618eb8324f9ed688fdbf09debf21a/platform/darwin/src/MGLMapSnapshotter.mm#L201-L210

would make sense?

Reply:
It looks like Android sends it in the initializer too: https://github.com/mapbox/mapbox-gl-native/pull/13475

Which should it be? /cc @tobrun

Reply:
@julianrex  I think the behavior should consist with `MGLMapview` did.

https://github.com/mapbox/mapbox-gl-native/blob/8bc362e3a2836cfa96ef14a41a0e882a1d08e881/platform/ios/src/MGLMapView.mm#L629-L632

The turnstile events send in the initializer of mapview.

Reply:
@julianrex @tobrun What's your opinion? Can I merge this PR?